src/netcdf.mk

```diff
@@ -32,14 +32,10 @@ define $(PKG)_BUILD
         --enable-netcdf-4 \
         --enable-hdf4 \
         --prefix='$(PREFIX)/$(TARGET)' \
-        CPPFLAGS="-D_DLGS_H" \
+        CPPFLAGS="-DWIN32_LEAN_AND_MEAN -DH4_BUILT_AS_STATIC_LIB=1" \
```

Comment:
OK, I split this pull request into to new ones, the first with imagemagick, openexr, dcmtk (should be uncontroversial) and the second one with portablexdr, hdf4, netcdf, gdal (which depend on one another). In the new version, H4_BUILT_AS_STATIC_LIB is added to the hdf4 header h4config.h, which is included by all hdf4 headers.